pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -19,255 +19,409 @@
  *   </li>
  *   <li>
- *     "/tables/{tableName}/segments/reload":
- *     Reload all segments of the table
+ *     Deprecated APIs:
+ *     <ul>
+ *       <li>"GET /tables/{tableName}/segments"</li>
+ *       <li>"GET /tables/{tableName}/segments/metadata"</li>
+ *       <li>"GET /tables/{tableName}/segments/crc"</li>
+ *       <li>"GET /tables/{tableName}/segments/{segmentName}"</li>
+ *       <li>"GET /tables/{tableName}/segments/{segmentName}/metadata"</li>
+ *       <li>"GET /tables/{tableName}/segments/{segmentName}/reload"</li>
+ *       <li>"POST /tables/{tableName}/segments/{segmentName}/reload"</li>
+ *       <li>"GET /tables/{tableName}/segments/reload"</li>
+ *       <li>"POST /tables/{tableName}/segments/reload"</li>
+ *     </ul>
  *   </li>
  * </ul>
- *
- * {@inheritDoc}
  */
-
-@Api(tags = Constants.TABLE_TAG)
+@Api(tags = Constants.SEGMENT_TAG)
 @Path("/")
 public class PinotSegmentRestletResource {
-  public static Logger LOGGER = LoggerFactory.getLogger(PinotSegmentRestletResource.class);
-  public static final Response.Status BAD_REQUEST = Response.Status.BAD_REQUEST;
-  public static final Response.Status INTERNAL_ERROR = Response.Status.INTERNAL_SERVER_ERROR;
-
-  private static final long _offlineToOnlineTimeoutInseconds = 5L;
+  private static Logger LOGGER = LoggerFactory.getLogger(PinotSegmentRestletResource.class);
 
   @Inject
   PinotHelixResourceManager _pinotHelixResourceManager;
 
   @GET
-  @Path("tables/{tableName}/segments")
   @Produces(MediaType.APPLICATION_JSON)
-  @ApiOperation(value = "Lists metadata or toggles segment states", notes = "Toggles segment states if 'state' is specified in query param, otherwise lists metadata")
-  // TODO: when state is not specified, this method returns the map from server to segments instead of segment metadata
-  public String toggleStateOrListMetadataForAllSegments(
+  @Path("/segments/{tableName}")
+  @ApiOperation(value = "Lists all segments", notes = "List all segments")
+  public List<Map<TableType, List<String>>> getSegments(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
-      @ApiParam(value = "enable|disable|drop", required = false) @QueryParam("state") String stateStr,
-      @ApiParam(value = "realtime|offline", required = false) @QueryParam("type") String tableTypeStr) {
-    CommonConstants.Helix.TableType tableType = Constants.validateTableType(tableTypeStr);
-    StateType state = Constants.validateState(stateStr);
+      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr) {
+    List<String> tableNamesWithType =
+        getExistingTableNamesWithType(tableName, Constants.validateTableType(tableTypeStr));
+    List<Map<TableType, List<String>>> resultList = new ArrayList<>(tableNamesWithType.size());
+    for (String tableNameWithType : tableNamesWithType) {
+      TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableNameWithType);
```

Comment:
If the table type is specified in the input, then we should list only the segments of that table type

Reply:
`getExistingTableNamesWithType` will only return the existing tables that match the given table name and type